### PR TITLE
Minor improvements in Makefiles

### PR DESCRIPTION
**Pull request by JainTwinkle**

### CODE REVIEW

File: mpi-proxy-split/Makefile_config.in [hunk 1]
```diff
@@ -37,7 +37,7 @@ else
   MPICXX = mpic++ -std=c++14
   MPIFORTRAN = mpifort
   MPIRUN = mpirun
-  MPI_LD_FLAG = -L$$HOME/mpich-static/usr/lib64 -lmpi -llzma -lz -lm -lxml2
+  MPI_LD_FLAG = -L$$HOME/mpich-static/usr/lib64 -lmpi -L$$HOME/local_install/lib -llzma -lz -lm -lxml2
```
Comment (gc00) [inline mpi-proxy-split/Makefile_config.in:40]: I suggest a FIXME comment above this line, to indicate that $$HOME/local_install/lib is a hard-wired pathname, and we should eventually provide a way to configure this pathame in configure-mana.

QUESTION: Are liblzma.a, libz.a, libm.a and libxml2.a the only extra libraries needed by static MPICH?  This doesn't affect the current PR, but I want to update the documentation in lower-half/README.mpich-static.  Thanks.

**Review by gc00 — APPROVED** [review]
I'm approving this in advance.  But please see my suggestion for an added comment.

QUESTION:  With this PR, can we build with MANA/MPICH on Discovery out-of-the-box?  If so, I'll update the documentation in lower-half/README.mpich-static.  Thanks.

**Comment by gc00** [comment]
@JainTwinkle , and please see one more addition for this PR, found in Issue #365.

File: mpi-proxy-split/Makefile_config.in (hunk 1 above)
Comment (JainTwinkle) [reply mpi-proxy-split/Makefile_config.in:37]: Added a FIXME.

Yes, these are the only libraries required on discovery when using the default mpich module (i.e., `mpich/3.3.2`).

**Comment by JainTwinkle** [comment]
> I'm approving this in advance. But please see my suggestion for an added comment.
> 
> QUESTION: With this PR, can we build with MANA/MPICH on Discovery out-of-the-box? If so, I'll update the documentation in lower-half/README.mpich-static. Thanks.

For discovery, MANA's installation documentation for CentOS should be sufficient. Moreover, the default module on discovery does provide a static mpi library. So, one can skip the mpich installation step on discovery and follow the rest.

**Comment by gc00** [comment]
@JainTwinkle wrote:
> For discovery, MANA's installation documentation for CentOS should be sufficient. Moreover, the default module on discovery does provide a static mpi library. So, one can skip the mpich installation step on discovery and follow the rest.

Good point.  Please see my two questions below.
I'd like to update our documentation for users at other sites who want to build MANA with MPICH or Open MPI.

**QUESTION 1:** So, on Discovery, if the mpich module is loaded, does `./configure-mana && make -j mana` automatically detect libmpi.a for mpich, and use it in lh_proxy?

@rajatpratapbisht and @leonidbelyaev , Do you have experience building a static libmpich.a somewhere (maybe on 'login' for Khoury (login.ccs.neu.edu)?

Could you briefly examine:  [README.mpich-static](https://github.com/mpickpt/mana/tree/main/mpi-proxy-split/lower-half#readme) or equivalently [README.mpich-static alt](https://github.com/mpickpt/mana/blob/main/mpi-proxy-split/lower-half/README.mpich-static) ?

**QUESTION 2:**  If we build a static libmpi.a for Open MPI, does the rest of MANA build out-of-the-box (with a suitable '-Ldir' or 'LD_LIBRARY_PATH=XXX') ?

**Comment by JainTwinkle** [comment]
> @JainTwinkle , For Question 1 (MPICH): you wrote:
> 
> > Yes, make does detect libmpi.a. But, it requires extra static libraries. If one follows the installation document, then those would be present in local_install/lib.
> 
> And we list three extra libraries that are needed. Is this a complete list? I want to document for CentOS anywhere, and not just for Discovery/CentOS. After this workshop paper, I "go on the road" to get other sites to use MANA.
> 
> Thanks.

@gc00 It depends on the MPICH version and how it is configured and built. I am not sure how to claim that this is a complete list. But generally, yes, these are the libraries we need extra. I guess we'll have to update the document as people test on different platforms in the future and provide feedback.

**Comment by leonidbelyaev** [comment]
Something else I've just found is that `./configure-mana` appears to be necessary to build working Makefiles on Discovery, but on Perlmutter, just `./configure` was sufficient.

If you build on a `./configure` on Discovery, you will get errors about a missing `mpicc` (something we found in our work and solved by doing manually).

Because `./configure-mana` also enables debug by default, the result is that there is no built-in way to obtain a working configuration for a production build of MANA on Discovery.

So, beyond changing `./configure-mana` to `./configure-developer`, it may also be desirable to create an additional `./configure-benchmark`, or the like.

**Comment by gc00** [comment]
@JainTwinkle , Thanks!  That helps a lot.
You wrote:
> @gc00 It depends on the MPICH version and how it is configured and built. I am not sure how to claim that this is a complete list. But generally, yes, these are the libraries we need extra. I guess we'll have to update the document as people test on different platforms in the future and provide feedback.

Thant's a key point:  What extra libraries you need depend on how the static MPICH was configured.  I'll add a comment to that effect.

**Review by gc00 — APPROVED** [review]
I'm approving this in advance.

However, there is a consensus that we should rename `configure-mana` to `configure-developer`.  You could add one more commit to do this name change.  But if you prefer, you can leave the extra commit for a different PR.

**Comment by leonidbelyaev** [comment]
I just performed an experiment on Discovery to compare the effect of running a MANA configured with `./configure-mana` (and thus, --enable-debug in `./configure`) and with just `./configure` (without).

The effect appears very significant! For the same lulesh workload, we have:

Without debug: `[146.24, 164.78, 170.58]`
With debug: `[302, 334, 450]`

Therefore, it is very important there is no confusion here. Perhaps a seperate `./configure-prod` or `./configure-benchmark` should also be added as required.

**Comment by gc00** [comment]
@leonidbelyaev , @JainTwinkle :  I agree that making it easy to run MANA on Discovery without the debugging mode is important.  Let's open a new PR for that, so we don't delay the improvements here.

I think there are two points here:
1.  As a temporary hack, you can just remote the `--enable-debug` flag inside the 'configure-mana` script, and then configure again.  This works for now.  (And probably you already knew this.)
2.  We can look at the configure flags in `configure-mana`, and update 'configure.ac' to use those flags by default on CentOS (but without the '--enable-debug' flag).  If you feel confident on modifying configure.ac, you can go ahead and do that.  Otherwise, I can do it fairly quickly, tonight.

**Comment by JainTwinkle** [comment]
@gc00 @leonidbelyaev 

The initial purpose of configure-mana was to have an easy and convenient script for developers with many MPI-related configure options. We already have a note in the script that one should modify parameters according to their environment. And we kept the `--enable_debug` option for developers. One can and should modify this file according to their usecase. I don't agree with having many versions of the configuration file.
We can do either of the two things:
1) Keep the --enable-debug and shout out that the debug option is enabled and please disable/remove it if required.
2) Remove the flag and let developers pass this one flag explicitly.

Please share your thoughts!

**Pull request merged**